pandas/tests/resample/test_base.py

```diff
@@ -118,6 +118,25 @@ def test_resample_empty_series(freq, empty_series, resample_method):
     assert_series_equal(result, expected, check_dtype=False)
 
 
+@pytest.mark.parametrize("freq", ["M", "D", "H"])
+@pytest.mark.parametrize("resample_method", ["count", "size"])
+def test_resample_count_empty_series(freq, resample_method):
+    # GH28427
+
+    empty_series = pd.Series([], dtype="datetime64[ns]", index=pd.DatetimeIndex([]))
+    result = getattr(empty_series.resample(freq), resample_method)()
+
+    expected = pd.Series([], dtype="int64", index=pd.DatetimeIndex([]))
+    if isinstance(empty_series.index, PeriodIndex):
+        expected.index = empty_series.index.asfreq(freq=freq)
+    else:
+        expected.index = empty_series.index._shallow_copy(freq=freq)
+
+    assert_index_equal(result.index, expected.index)
+    assert result.index.freq == expected.index.freq
+    assert_series_equal(result, expected, check_dtype=True)
```

Comment:
I think can just use `assert_series_equal` here (would cover index check as well). Also you shouldn't need `check_dtype` as an argument as that is default